contracts/Uniswap/UniswapAnchoredView.sol

```diff
@@ -209,14 +217,15 @@ contract UniswapAnchoredView is UniswapConfig {
 
         // Calculate uniswap time-weighted average price
         FixedPoint.uq112x112 memory priceAverage = FixedPoint.uq112x112(uint224((nowCumulativePrice - oldCumulativePrice) / timeElapsed));
-        uint anchorPriceUnscaled = mul(priceAverage.decode112with18(), conversionFactor);
+        uint rawUniswapPriceMantissa = mul(priceAverage.decode112with18(), conversionFactor);
         uint anchorPrice;
 
-        // Adjust anchor price to val * 1e6 decimals format
+        // Adjust rawUniswapPrice according to the units of the non-ETH asset
+        // In the case of ETH, we would have to scale by 1e6 / USDC_UNITS, but since baseUnit2 is 1e6 (USDC), it cancels
         if (config.isUniswapReversed) {
-            anchorPrice = anchorPriceUnscaled / config.baseUnit;
+            anchorPrice = mul(rawUniswapPriceMantissa, ethBaseUnit) / config.baseUnit / expScale;
```

Comment:
This is a huge change too. How did this cancel out for markets like WBTC?

Reply:
not sure what you mean, i mean this to be a non-change. just showing the logic that `expScale` == `ethBaseUnit, though this could just as well be a comment instead of in the code itself, perhaps that would help overflow concerns

Reply:
Yea I agree as a comment would be better